apis/apps/v1alpha1/componentresourceconstraint_types.go

```diff
@@ -29,10 +29,37 @@ import (
 // ComponentResourceConstraintSpec defines the desired state of ComponentResourceConstraint
 type ComponentResourceConstraintSpec struct {
 	// Component resource constraints
+	// +patchMergeKey=name
+	// +patchStrategy=merge,retainKeys
+	// +listType=map
+	// +listMapKey=name
+	// +kbubebuilder:validation:Required
 	Constraints []ResourceConstraint `json:"constraints,omitempty"`
+
+	// selector is used to bind the component resource constraint to a component.
+	// +optional
+	Selector []ComponentResourceConstraintSelector `json:"selector,omitempty"`
```

Comment:
as listType=map and have clusterDefRef as key?

Reply:
fixed